evoting/lib/transaction.go

```diff
@@ -79,31 +75,24 @@ func NewTransaction(data interface{}, user uint32, signature []byte) *Transactio
 	return transaction
 }
 
-// Digest appends the digits of sciper to master genesis skipblock ID
-func (t *Transaction) Digest(s *skipchain.Service, genesis skipchain.SkipBlockID) []byte {
-	var message []byte
-	switch {
-	case t.Master != nil:
-		message = t.Master.ID
-	case t.Election != nil:
-		message = t.Election.Master
-	default:
-		election, _ := GetElection(s, genesis, false, t.User)
-		if election == nil {
-			return nil
-		}
-		message = election.Master
+// Sign generates a signature on the marshaled form of the txn,
+// without a Signature, then puts that signature into the txn.
+func (t *Transaction) Sign(priv kyber.Scalar) error {
+	t.Signature = nil
+	data, err := protobuf.Encode(t)
```

Comment:
I would somehow add something more to the signature because if someone votes twice, an attacker could potentially replay the first vote to erase the last one, right ?

Reply:
hmm, how about the block index? seems doable, but will cause me to re-refactor the code.